### ISSUE TITLE
3D condition and Geometry deformation

### ISSUE THREAD

**Issue by ju1987yetchung** [junfeng-chen/position_induced_transformer#1]
Hi, author
could you tell me is this possible to implement on 3D case and if the geometry is not a block or square?
thank you a lot!

**Comment by AI4Equations** [comment]
> Hi, author could you tell me is this possible to implement on 3D case and if the geometry is not a block or square? thank you a lot!

Yes, it's possible to apply PiT and position-attention to 3D cases with arbitrary geometries, such as 3D car simulations, as long as the data are transformed into point clouds.

However, be mindful of memory usage and training time, especially with large 3D datasets.

**Comment by junfeng-chen** [comment]
> Hi, author could you tell me is this possible to implement on 3D case and if the geometry is not a block or square? thank you a lot!

Yes, it's possible to apply PiT and position-attention to 3D cases with arbitrary geometries, such as 3D car simulations, as long as the data are transformed into point clouds.

However, be mindful of memory usage and training time, especially with large 3D datasets.
